sovereign-forge/tests/sovereign_forge_blackbox_setup.rs

```diff
@@ -47,6 +48,30 @@ impl SovereignForgeTestState {
         self.register_token_handler(3, TOKEN_HANDLER_SC_ADDRESS, None);
     }
 
+    pub fn deploy_template_scs(&mut self, templates: Option<Vec<ScArray>>) {
+        for sc in templates.unwrap_or_default().into_iter() {
+            match sc {
+                ScArray::ChainConfig => {
+                    self.common_setup
+                        .deploy_chain_config(OptionalValue::None, None);
+                }
+                ScArray::ESDTSafe => {
+                    self.common_setup.deploy_mvx_esdt_safe(OptionalValue::None);
+                }
+                ScArray::FeeMarket => {
+                    self.common_setup.deploy_fee_market(None, ESDT_SAFE_ADDRESS);
+                }
+                ScArray::HeaderVerifier => {
+                    self.common_setup.deploy_header_verifier(vec![]);
+                }
+                ScArray::ChainFactory => {
+                    self.common_setup.deploy_chain_factory();
```

Comment:
`chain-factory` is not a template SC. It is deployed in all shard before everything else